### PR TITLE
handle member signed in memeber service

### PR DESCRIPTION
**Pull request by FredrikAreschoug**

### CODE REVIEW

File: src/main/kotlin/com/hedvig/underwriter/model/Quote.kt [hunk 1]
```diff
@@ -147,6 +150,88 @@ data class Quote(
...
+            is ApartmentData -> CalculateQuoteRequestDto(
+                memberId = memberId!!,
+                ssn = data.ssn!!,
+                firstName = data.firstName!!,
+                lastName = data.lastName!!,
+                birthDate = data.ssn!!.birthDateFromSsn(),
+                student = data.isStudent,
+                address = Address(
+                    data.street!!,
+                    data.city!!,
```
Comment (meletios) [inline src/main/kotlin/com/hedvig/underwriter/model/Quote.kt]: In some cases, we are missing the city, are we going to have it always in underwritter?

File: src/main/kotlin/com/hedvig/underwriter/model/Quote.kt [hunk 2]
```diff
@@ -147,6 +150,88 @@ data class Quote(
...
+            is HouseData -> CalculateQuoteRequestDto(
+                memberId = memberId!!,
+                ssn = data.ssn!!,
+                firstName = data.firstName!!,
+                lastName = data.lastName!!,
+                birthDate = data.ssn!!.birthDateFromSsn(),
+                student = false,
+                address = Address(
+                    data.street!!,
+                    data.city!!,
```
Comment (meletios) [inline src/main/kotlin/com/hedvig/underwriter/model/Quote.kt]: same here

File: src/main/kotlin/com/hedvig/underwriter/model/Quote.kt [hunk 3]
```diff
@@ -147,6 +150,88 @@ data class Quote(
...
+                livingSpace = data.livingSpace!!.toFloat(),
+                houseType = ProductPricingProductTypes.HOUSE,
+                currentInsurer = currentInsurer,
+                personsInHouseHold = data.householdSize!!,
+                ancillaryArea = data.ancillaryArea,
+                yearOfConstruction = data.yearOfConstruction,
+                numberOfBathrooms = data.numberOfBathrooms,
+                extraBuildings = data.extraBuildings?.map { extraBuilding ->
+                    ExtraBuildingDTO(
+                        type = when (extraBuilding.type) {
```
Comment (palmenhq) [inline src/main/kotlin/com/hedvig/underwriter/model/Quote.kt]: `com.hedvig.underwriter.graphql.type.ExtraBuildingType.valueOf(extraBuliding.type.toString())` would be slightly hacky but also way easier to work with

File: src/main/kotlin/com/hedvig/underwriter/service/QuoteServiceImpl.kt [hunk 4]
```diff
@@ -233,17 +239,38 @@ class QuoteServiceImpl(
...
 
-        if (quoteWithMember.data is PersonPolicyHolder<*>) {
-            memberService.updateMemberSsn(memberId.toLong(), UpdateSsnRequest(ssn = quoteWithMember.data.ssn!!))
+        return Right(signQuoteWithMemberId(quoteWithMember, false, body.email))
+    }
+
+    override fun memberSigned(memberId: String) {
+        quoteRepository.findOneByMemberId(memberId)?.let { quote ->
+            signQuoteWithMemberId(quote, true, null)
+        } ?: throw IllegalStateException("Preformed member sign with no quote!")
```
Comment (palmenhq) [inline src/main/kotlin/com/hedvig/underwriter/service/QuoteServiceImpl.kt]: typo

File: src/main/kotlin/com/hedvig/underwriter/service/QuoteServiceImpl.kt [hunk 5]
```diff
@@ -233,17 +239,38 @@ class QuoteServiceImpl(
...
+    }
+
+    private fun signQuoteWithMemberId(
+        quote: Quote,
+        signedInMemberService: Boolean,
+        email: String?
+    ): SignedQuoteResponseDto {
+        checkNotNull(quote.memberId) { "Quote must have a member id!" }
```
Comment (palmenhq) [inline src/main/kotlin/com/hedvig/underwriter/service/QuoteServiceImpl.kt]: maybe we could add a quote id or something here? Just so we can actually track down what happened here

File: src/main/kotlin/com/hedvig/underwriter/web/QuoteController.kt [hunk 6]
```diff
@@ -112,4 +112,10 @@ class QuoteController @Autowired constructor(
     fun getAllQuotesFromMemberId(@PathVariable memberId: String): ResponseEntity<List<QuoteDto>> {
         return ResponseEntity.ok(quoteService.getQuotesForMemberId(memberId))
     }
+
+    @PostMapping("/member/{memberId}/signed")
+    fun memberSigned(@PathVariable memberId: String): ResponseEntity<String> {
+        quoteService.memberSigned(memberId)
+        return ResponseEntity.ok("")
```
Comment (palmenhq) [inline src/main/kotlin/com/hedvig/underwriter/web/QuoteController.kt]: `ResponseEntity.noContent()`?

File: src/main/kotlin/com/hedvig/underwriter/service/QuoteServiceImpl.kt [hunk 7]
```diff
@@ -233,17 +239,38 @@ class QuoteServiceImpl(
...
+        return Right(signQuoteWithMemberId(quoteWithMember, false, body.email))
+    }
+
+    override fun memberSigned(memberId: String) {
+        quoteRepository.findOneByMemberId(memberId)?.let { quote ->
+            signQuoteWithMemberId(quote, true, null)
+        } ?: throw IllegalStateException("Preformed member sign with no quote!")
+    }
+
+    private fun signQuoteWithMemberId(
```
Comment (meletios) [inline src/main/kotlin/com/hedvig/underwriter/service/QuoteServiceImpl.kt]: Shall we merge this with the signQuote method? 🤔

Comment (meletios) [reply src/main/kotlin/com/hedvig/underwriter/service/QuoteServiceImpl.kt]: It seems that they kind of do similar things

Comment (FredrikAreschoug) [reply src/main/kotlin/com/hedvig/underwriter/service/QuoteServiceImpl.kt]: `signQuote` returns `Right(signQuoteWithMemberId(quoteWithMember, false, body.email))`

Comment (meletios) [reply src/main/kotlin/com/hedvig/underwriter/service/QuoteServiceImpl.kt]: Hahaha thats the thing that i was thinking about. You already did that. You can discard my comment then 😅

**Pull request merged**